python-threatexchange/README.md

```diff
@@ -27,113 +19,108 @@ If you don't have `pip`, learn how to install it [here](https://pip.pypa.io/en/s
 $ python3 -m pip install threatexchange [--upgrade]
 ```
 
-The base installation includes matching for:
+# Introduction
+Trust and safety is a generally hard problem. An issue that makes the problem harder is that most platforms attempt to keep their platforms safe on their own, despite bad actors and viral content spreading from platform to platform. This results not only in duplicate effort in building out technical capability to detect harmful content, but also duplicate effort in preventing the spread of known harmful content, since each platform is fighting potentially the same copies of content on their own.
 
-**Text**
-* Raw Text
-* Trend Queries
+One technique that can allow platforms to combine efforts to combat harm is by sharing signatures of content that they have already detected on their own, or the inputs to various trust and safety tools that can be used to find harmful content. The most well-known are photo/video hash sharing programs like those operated by the National Center for Exploited and Missing Children (NCMEC) and the Global Internet Forum to Counter Terrorism. However, there are also lesser-known exchanges as well, such as StopNCII.org and Meta's ThreatExchange platform.
 
-**URL**
-* URLs (simple)
+The python-threatexchange library aims to simplify the exchange of signals via platforms like the above, as well as provide a baseline of functionality available to simplify the testing and creation of new exchanges and techniques, as well as provide cross-compatibility. 
 
-**Photo**
-* MD5
+## Philisophy of the Library
+Trust and safety is not owned by Meta. However, this library is maintained by a small team at Meta with a limited range of experience, and so we'll probably prioritize the usecases we are most familiar with. We believe that accessibility is a barrier for many platforms and so will put as much as we can in the open. We also understand that it may not make sense to use only publicly visible approaches, and welcome platform-specific modifications and derivatives. However, we also accept pull requests! If you think functionality is generally applicable, or you have a bug bothering you, we accept pull requests! If you are thinking a larger change may be needed (such as adding an entirely new subcommand to the CLI), we definitely appreciate if you reach out to talk through a feature before submitting it!
 
-**Video**
-* MD5
+## General Expectation for Compatibility and Versioning
+1. Major versions (1.X.X => 2.0.0) are not guaranteed to backwards compatible. However, post 1.0.0, tooling to try and migrate state forward will be available. Please reach out if you need help migrating forward. 
```

Comment:
this is speaking to a real or abstract versions, if abstract we should add "e.g." to the parentheses to make it clear we are not talking about specific pytx versions.
Also nit: "will" -> "should" to cover our bases (thinking of the case where a bug is discovered that makes such tooling impossible)

Reply:
We talked about this offline, 

I believe that we "will" have tooling to "try" to migrate. We cannot provide no solution. Barrett counters that this should be a team decision on if we will uphold this, which I think narrows to just @BarrettOlson and @schatten.

Are you two in favor to keeping breaking changes to major version numbers, and to provide a route for backwards compatibility for state between major versions?